src/webpack/index.js

```diff
@@ -0,0 +1,50 @@
+const createHash = require('webpack/lib/util/createHash');
+
+const NAME = 'RemixxStaticModuleIdsPlugin'
+
+class RemixxStaticModuleIdsPlugin {
+  constructor(options) {
+    if (!options) options = {};
+
+
+    this.options = Object.assign(
+      {
+        context: null,
+        hashFunction: 'md4',
+        hashDigest: 'base64',
+        hashDigestLength: 4
+      },
+      options
+    );
+  }
+
+  apply(compiler) {
+    const options = this.options;
+    compiler.hooks.compilation.tap(NAME, compilation => {
+      const usedIds = new Set();
+      compilation.hooks.beforeModuleIds.tap(
+        NAME,
+        modules => {
+          for (const module of modules) {
+            if (module.id === null && module.libIdent) {
+              const id = module.libIdent({
+                context: this.options.context || compiler.options.context
+              });
+              const hash = createHash(options.hashFunction);
+              hash.update(id);
+              const hashId = hash.digest(options.hashDigest);
+              let len = options.hashDigestLength;
+              while (usedIds.has(hashId.substr(0, len))) len++;
```

Comment:
1) is this just a safety precaution, in case the 4 character hash happens to exist already? 

2) what does `module.libIdent` do? 
is the idea that `beforeModuleIds` gives us an opportunity to define them ourselves? ...I assume u saw how this was done elsewhere and copied it for our purposes--link?

3) why is all this has generation code more than one line? I mean i assume that's the pattern the other webpack libraries are using. Im just curious why it's not just like `generate8CharacterHash(fileName)` and done??

----

overall, me likey. ..4) is there a reason we want to take over defining module IDs rather than wait until the next compilation hook, and just take the module ID they give us? 

----

5) crap, why are we setting the module ID? we are supposed to be setting the chunk id. The module ID i assume is a unique ID for each file. We want the ID for the resulting chunk.